File: PythonNCAA/scripts/_verify_lines.mjs
#!/usr/bin/env node
import fs from "fs";

const store = JSON.parse(fs.readFileSync("data/history.json", "utf8"));
const runs = store.runs || [];

// Find games with big lines and check if the sign convention makes sense
console.log("=== Games with |line| > 15 (first 20) ===");
let count = 0;
for (const r of runs) {
  for (const g of r.games) {
    if (count >= 20) break;
    const absLine = Math.abs(g.line);
    if (absLine < 15) continue;
    if (g.homeScore === undefined) continue;

    const margin = g.homeScore - g.awayScore;
    console.log(`${r.date} | ${g.away} @ ${g.home} | line=${g.line} | score: away=${g.awayScore} home=${g.homeScore} | homeMargin=${margin}`);

    // If line > 0 (home dog), the favorite is away.
    // Favorite should usually win by more than the line. Check this.
    if (g.line > 0) {
      // Home is dog. Away is fav. Away "should" win.
      // If away won: margin < 0 (home lost)
      // Dog covers if |margin| < line, i.e. they didn't lose by more than the spread
      // Or dog won outright
      const dogCovered = margin + g.line > 0;
      console.log(`  Home is ${g.line} pt dog. Away won by ${-margin}. Dog covered? ${dogCovered}`);
    } else {
      // Away is dog. Home is fav.
      const dogCovered = -(margin + g.line) > 0;
      console.log(`  Away is ${absLine} pt dog. Home won by ${margin}. Dog covered? ${dogCovered}`);
    }
    count++;
  }
}

// KEY CHECK: what % of ALL dogs cover in NCAA? Should be ~50%
let totalDogW = 0, totalDogL = 0;
for (const r of runs) {
  for (const g of r.games) {
    if (g.homeScore === undefined || g.awayScore === undefined) continue;
    if (g.status === "SKIPPED" || g.status === "MISSING_STATS") continue;
    if (!g.line || g.line === 0) continue;

    const margin = g.homeScore - g.awayScore;
    const v = margin + g.line;

    // Dog is home when line > 0, away when line < 0
    if (g.line > 0) {
      // home is dog, covers when v > 0
      if (v > 0) totalDogW++; else if (v < 0) totalDogL++;
    } else {
      // away is dog, covers when v < 0
      if (v < 0) totalDogW++; else if (v > 0) totalDogL++;
    }
  }
}

console.log(`\n=== ALL dogs ATS (no model filter): ${totalDogW}-${totalDogL} (${(totalDogW/(totalDogW+totalDogL)*100).toFixed(1)}%) ===`);
console.log("(Should be ~50% if lines/grading are correct)");

File: PythonNCAA/scripts/_verify_lines.py
#!/usr/bin/env python3
# scripts/_verify_lines.py — Verify line sign convention
import json

store = json.loads(open("data/history.json").read())
runs = store.get("runs") or []

print("=== Games with |line| > 15 (first 20) ===")
count = 0
for r in runs:
    for g in (r.get("games") or []):
        if count >= 20: break
        abs_line = abs(g.get("line", 0))
        if abs_line < 15: continue
        if g.get("homeScore") is None: continue
        margin = g["homeScore"] - g["awayScore"]
        print(f"{r.get('date')} | {g['away']} @ {g['home']} | line={g['line']} | score: away={g['awayScore']} home={g['homeScore']} | homeMargin={margin}")
        if g["line"] > 0:
            dog_covered = margin + g["line"] > 0
            print(f"  Home is {g['line']} pt dog. Away won by {-margin}. Dog covered? {dog_covered}")
        else:
            dog_covered = -(margin + g["line"]) > 0
            print(f"  Away is {abs_line} pt dog. Home won by {margin}. Dog covered? {dog_covered}")
        count += 1

total_dog_w, total_dog_l = 0, 0
for r in runs:
    for g in (r.get("games") or []):
        if g.get("homeScore") is None or g.get("awayScore") is None: continue
        if g.get("status") in ("SKIPPED", "MISSING_STATS"): continue
        if not g.get("line") or g["line"] == 0: continue
        margin = g["homeScore"] - g["awayScore"]
        v = margin + g["line"]
        if g["line"] > 0:
            if v > 0: total_dog_w += 1
            elif v < 0: total_dog_l += 1
        else:
            if v < 0: total_dog_w += 1
            elif v > 0: total_dog_l += 1

print(f"\n=== ALL dogs ATS (no model filter): {total_dog_w}-{total_dog_l} ({total_dog_w / (total_dog_w + total_dog_l) * 100:.1f}%) ===")
print("(Should be ~50% if lines/grading are correct)")
